lib/OpenStackExtract/MiqOpenStackVm/MiqOpenStackCommon.rb

```diff
@@ -50,6 +53,55 @@ def get_image_file_glance_v2(image_id)
     end
     tf
   end
+  
+  def create_image_from_snapshot(snapshot_id, disk_format)
+    snapshot = volume_service.snapshots.get(snapshot_id)
+    volume_options = {
+      :name        => "Temp Volume from #{snapshot.name}",
+      :size        => snapshot.size,
+      :snapshot_id => snapshot_id
+    }
+    volume = volume_service.volumes.new(volume_options)
+    volume.save
+    while volume.status != 'available'
+      sleep(10)
+      volume = volume_service.volumes.get(volume.id)
+    end
+    response = volume_service.action(volume.id, 'os-volume_upload_image' => {
+                                       :image_name => "Temp Image from #{snapshot.name}",
+                                       :disk_format => disk_format})
+    image_id = response.body["os-volume_upload_image"]["image_id"]
+    while image_service.images.get(image_id).status.downcase != 'active'
+      sleep(10)
+    end
+    return image_service.images.get(image_id)
+  ensure
+    volume.destroy
+  end
+  
+  def get_image_file_glance_v2(image_id)
+    log_prefix = "#{self.class.name}##{__method__}"
+
+    image = image_service.images.get(image_id)
+    raise "Image #{image_id} not found" unless image
+    $log.debug "#{log_prefix}: image = #{image.class.name}"
+
+    if image.size.to_i == 0
+      # try getting image from metadata; oddly the image metadata is only available if
+      # image is queried through compute service
+      if snapshot_id = get_image_metadata_snapshot_id(compute_service.images.get(image_id))
+        begin
+          temp_image = create_image_from_snapshot(snapshot_id, image.disk_format)
```

Comment:
@tzumainn 
The `begin` block should start after the `temp_image =` line, because if `create_image_from_snapshot` fails, `temp_image` won't be set in the `ensure` block.